klasses/models.py

```diff
@@ -82,6 +85,34 @@ def payment_deadline(self):
         """
         return self.installment_set.aggregate(payment_deadline=models.Max('deadline'))['payment_deadline']
 
+    @property
+    def next_installment(self):
+        """
+        Get the next installment
+        """
+        return self.installment_set.exclude(
+            deadline=None).filter(deadline__gte=datetime.datetime.now(tz=pytz.UTC)).first()
+
+    @property
+    def next_payment_deadline_days(self):
+        """
+        Returns the number of days until the next payment is due
+        """
+        if self.next_installment is None:
```

Comment:
if the query is exactly the same, it should be cached, but I'll add the extra variable